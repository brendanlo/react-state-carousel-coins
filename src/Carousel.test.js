import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";

it("smoke test: Carousel component", function () {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
});

it("works when you click on the right arrow", function () {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".fa-chevron-circle-right");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  // CR: always be extra careful around not.ToBe.. could be a false positive
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();
});

it("works when you click on the left arrow", function () {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );

  // move forward in the carousel
  const rightArrow = container.querySelector(".fa-chevron-circle-right");
  fireEvent.click(rightArrow);

  // move backward in the carousel
  const leftArrow = container.querySelector(".fa-chevron-circle-left");
  fireEvent.click(leftArrow);


  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

});


it("left arrow doesn't show up on first image", function () {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );

  //CR for extra comfort, the right arrow should be here so ALSO check on this
  // in your testing too
  const leftButton = container.querySelector(".fa-chevron-circle-left");
  expect(leftButton).not.toBeInTheDocument();

});

it("right arrow doesn't show up on the last image", function () {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // move forward in the carousel 2 positions to the right
  const rightArrow = container.querySelector(".fa-chevron-circle-right");
  fireEvent.click(rightArrow);
  fireEvent.click(rightArrow);

  const leftButton = container.querySelector(".fa-chevron-circle-right");
  expect(leftButton).not.toBeInTheDocument();
});

it("is a snapshot of a working version of Carousel", function () {
  const { container, debug } = render(
    <Carousel
      photos={[
        {
          src: "image1",
          caption: "Photo by Richard Pasquarella on Unsplash"
        }
      ]}
      title="Shells from far-away beaches"
    />);

  expect(container).toMatchSnapshot();
});