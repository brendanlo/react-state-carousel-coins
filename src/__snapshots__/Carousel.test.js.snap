// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is a snapshot of a working version of Carousel 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Shells from far-away beaches
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="fas fa-chevron-circle-left fa-2x"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Photo by Richard Pasquarella on Unsplash
        </h4>
        <img
          alt="Photo by Richard Pasquarella on Unsplash"
          class="Card-image"
          src="image1"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          1
          .
        </small>
      </div>
      <i
        class="fas fa-chevron-circle-right fa-2x"
      />
    </div>
  </div>
</div>
`;
